### PR TITLE
fix(components/header): fix footer overlapping old header sidemenu

### PR DESCRIPTION
**Pull request by alexandereneroth**
**Describe pull-request**  
- In @scania/components: Fixes old footer overlapping side menu.
- In angular sandbox: Changes header so that the bug can be reproduced.

**Solving issue**  
Fixes: #257, https://tegel.atlassian.net/browse/DTS-787 

**How to test**  
1. Checkout the branch locally
1. Make sure you are on node version 16 and you ran npm install in the appropriate folders
1. Go to the sdds root folder
1. Run this and it should hopefully just work™️
```
cd ./components && npm run build && npm link && cd ../sandbox/angular && npm link @scania/components && npm start
```
5. Once the dev server is running in the angular sandbox, go to the path `/short` in the URL bar
6. Open the sidemenu and see it overlapping the footer! 🏁 

**Screenshots**  
Previous issue:
Screenshot 2023-04-26 at 17 03 42

After fix:
Screenshot 2023-04-26 at 17 04 37

**Additional context**  
Could not reproduce the original issue exactly but it is pretty old. This fix should fix it nonetheless.

### CODE REVIEW

File: components/src/components/header/header.scss [hunk 1]
```diff
@@ -39,11 +37,12 @@ html,
   height: 64px;
   width: 100%;
   z-index: 3;
+  position: relative;
```
Comment (mJarsater) [inline components/src/components/header/header.scss:40]: Is this the only "real" change?

Comment (mJarsater) [reply components/src/components/header/header.scss:40]: If so, im not sure I understand how that fixes our issue. Would love a quick walkthrough

Comment (alexandereneroth) [reply components/src/components/header/header.scss:40]: Its basically this: z-index doesnt do anything on position:static (which is the default) 😃 
If you read here, you'll see its supposed to be set on a "positioned" element: https://developer.mozilla.org/en-US/docs/Web/CSS/z-index

Comment (mJarsater) [reply components/src/components/header/header.scss:40]: Aaah! Lovely!

**Review by mJarsater — APPROVED** [review]
🚀

**Pull request merged**